packages/js/components/src/flex-box/flex-box.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import classNames from 'classnames';
+import { unset } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+export type FlexStyleProps = {
+	flexDirection?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
+	flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
+	justifyContent?:
+		| 'flex-start'
+		| 'flex-end'
+		| 'center'
+		| 'space-between'
+		| 'space-around'
+		| 'space-evenly';
+	alignItems?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
+	alignContent?:
+		| 'flex-start'
+		| 'flex-end'
+		| 'center'
+		| 'space-between'
+		| 'space-around'
+		| 'space-evenly'
+		| 'stretch';
+	rowGap?: string | number;
+	columnGap?: string | number;
+	width?: string | number;
```

Comment:
For non-flex properties, could we just pass those in under a `style` prop?  How often do we anticipate using height and width?

Reply:
Yeah, that's better. I initially had height and width as full properties because I wanted intelligent defaults, i.e. if you're making a FlexBox with a flex-direction of 'row' you don't have to set any properties at all. But it's better to expose the style prop to the consumer.